tests/test_misc.py

```diff
@@ -1516,3 +1516,89 @@ def test_revault(revault_network, bitcoind, executor):
         wait_for(
             lambda: w.rpc.listvaults([], [deposit])["vaults"][0]["status"] == "canceled"
         )
+
+@pytest.mark.skipif(not POSTGRES_IS_SETUP, reason="Needs Postgres for servers db")
+def test_reorged_revault(revault_network, bitcoind):
+    revault_network.deploy(4, 2, csv=12)
+    stks = revault_network.stk_wallets
+    mans = revault_network.man_wallets
+    vault = revault_network.fund(32)
+    revault_network.secure_vault(vault)
+    revault_network.activate_vault(vault)
+    deposit = f"{vault['txid']}:{vault['vout']}"
+    amount = vault["amount"]
+
+    addr = bitcoind.rpc.getnewaddress()
+    feerate = 1
+    fee = revault_network.compute_spendtx_fees(feerate, 1, 1)
+    destinations = {addr: amount - fee}
+    revault_network.unvault_vaults([vault], destinations, feerate)
+    unvault_tx = mans[0].rpc.listonchaintransactions([deposit])[
+        "onchain_transactions"
+    ][0]["unvault"]
+
+    # Now let's cancel the spending
+    revault_network.cancel_vault(vault)
+
+    cancel_tx = mans[0].rpc.listonchaintransactions([deposit])[
+        "onchain_transactions"
+    ][0]["cancel"]
+
+    # Reorging, but not unconfirming the cancel
+    bitcoind.simple_reorg(cancel_tx["blockheight"])
+
+    for w in stks + mans:
+        w.wait_for_logs(
+            [
+                "Detected reorg",
+                f"Vault {deposit}'s Cancel transaction is still confirmed",
+                "Rescan of all vaults in db done.",
+            ]
+        )
+
+
+    # Let's unconfirm the cancel and check that the vault is now in 'canceling' state
+    bitcoind.simple_reorg(cancel_tx["blockheight"], shift=-1)
+
+    for w in stks + mans:
+        w.wait_for_logs(
+            [
+                "Detected reorg",
+                f"Vault {deposit}'s Cancel transaction .* got unconfirmed",
+                "Rescan of all vaults in db done.",
+            ]
+        )
+
+    for w in stks + mans:
+        wait_for(
+            lambda: w.rpc.listvaults([], [deposit])["vaults"][0]["status"] == "canceling"
+        )
+
+    # Confirming the cancel again
+    bitcoind.generate_block(6)
+
+    for w in stks + mans:
+        w.wait_for_log("Cancel tx .* was confirmed at height .*")
+        wait_for(
+            lambda: w.rpc.listvaults([], [deposit])["vaults"][0]["status"] == "canceled"
+        )
+
+    # Let's unconfirm the unvault
+    bitcoind.simple_reorg(unvault_tx["blockheight"], shift=-1)
+    for w in stks + mans:
+        w.wait_for_log(f"Vault {deposit}'s Unvault transaction .* got unconfirmed")
+
+    # Here we go canceling everything again
+    bitcoind.generate_block(6)
```

Comment:
Here too i think a single block should suffice with `wait_for_mempool=2`

Reply:
This would make the below a single loop where you just wait for the logs and wait for `canceled` status at the end

Reply:
Sigh, of course i can't have it to work. Will look into it, here is my diff (with the Python linter, btw)

```diff
diff --git a/tests/servers/coordinatord b/tests/servers/coordinatord
--- a/tests/servers/coordinatord
+++ b/tests/servers/coordinatord
@@ -1 +1 @@
-Subproject commit 7136b781b86eeb5be55d8a7b18071444afbaa6a6
+Subproject commit 7136b781b86eeb5be55d8a7b18071444afbaa6a6-dirty
diff --git a/tests/servers/cosignerd b/tests/servers/cosignerd
--- a/tests/servers/cosignerd
+++ b/tests/servers/cosignerd
@@ -1 +1 @@
-Subproject commit 52d7aeaec3bea9647b0d980c4e595f5cdcaaf4cf
+Subproject commit 52d7aeaec3bea9647b0d980c4e595f5cdcaaf4cf-dirty
diff --git a/tests/test_misc.py b/tests/test_misc.py
index 67b783c..e142d82 100644
--- a/tests/test_misc.py
+++ b/tests/test_misc.py
@@ -1453,6 +1453,7 @@ def test_large_spends(revault_network, bitcoind, executor):
         lambda: len(man.rpc.listvaults(["spent"], deposits)["vaults"]) == len(deposits)
     )
 
+
 @pytest.mark.skipif(not POSTGRES_IS_SETUP, reason="Needs Postgres for servers db")
 def test_revault(revault_network, bitcoind, executor):
     """
@@ -1466,35 +1467,46 @@ def test_revault(revault_network, bitcoind, executor):
     stks[0].wait_for_deposits([deposit])
 
     # Can't cancel an unconfirmed deposit
-    with pytest.raises(RpcError, match="Invalid vault status: 'funded'. Need 'unvaulting'"):
+    with pytest.raises(
+        RpcError, match="Invalid vault status: 'funded'. Need 'unvaulting'"
+    ):
         stks[0].rpc.revault(deposit)
 
     # A manager gets the same error: both parties can revault
-    with pytest.raises(RpcError, match="Invalid vault status: 'funded'. Need 'unvaulting'"):
+    with pytest.raises(
+        RpcError, match="Invalid vault status: 'funded'. Need 'unvaulting'"
+    ):
         man.rpc.revault(deposit)
 
     revault_network.secure_vault(vault)
 
     # Secured is not good enough though
-    with pytest.raises(RpcError, match="Invalid vault status: 'secured'. Need 'unvaulting'"):
+    with pytest.raises(
+        RpcError, match="Invalid vault status: 'secured'. Need 'unvaulting'"
+    ):
         stks[0].rpc.revault(deposit)
 
     revault_network.activate_vault(vault)
 
     # Active? Not enough!
-    with pytest.raises(RpcError, match="Invalid vault status: 'active'. Need 'unvaulting'"):
+    with pytest.raises(
+        RpcError, match="Invalid vault status: 'active'. Need 'unvaulting'"
+    ):
         stks[0].rpc.revault(deposit)
 
     # Now we want to broadcast the unvault tx without having an associated spend tx
     # First of all, we need the unvault psbt finalized
-    unvault_psbt = stks[0].rpc.listpresignedtransactions([deposit])['presigned_transactions'][0]['unvault']
-    unvault_tx = bitcoind.rpc.finalizepsbt(unvault_psbt)['hex']
+    unvault_psbt = stks[0].rpc.listpresignedtransactions([deposit])[
+        "presigned_transactions"
+    ][0]["unvault"]
+    unvault_tx = bitcoind.rpc.finalizepsbt(unvault_psbt)["hex"]
     bitcoind.rpc.sendrawtransaction(unvault_tx)
 
     # Unvaulting! And there's no associated spend tx! Is revault broken?
     for w in stks + [man]:
         wait_for(
-            lambda: w.rpc.listvaults([], [deposit])["vaults"][0]["status"] == "unvaulting"
+            lambda: w.rpc.listvaults([], [deposit])["vaults"][0]["status"]
+            == "unvaulting"
         )
 
     # Nah it's not, just broadcast the cancel
@@ -1504,19 +1516,21 @@ def test_revault(revault_network, bitcoind, executor):
     for w in stks + [man]:
         w.wait_for_log("Unvault transaction at .* is now being canceled")
         wait_for(
-            lambda: w.rpc.listvaults([], [deposit])["vaults"][0]["status"] == "canceling"
+            lambda: w.rpc.listvaults([], [deposit])["vaults"][0]["status"]
+            == "canceling"
         )
 
     bitcoind.generate_block(6, wait_for_mempool=1)
 
     # Funds are safe, we happy
     for w in stks + [man]:
-        wait_for(lambda: w.rpc.call("getinfo")["blockheight"]  == 113)
+        wait_for(lambda: w.rpc.call("getinfo")["blockheight"] == 113)
         w.wait_for_log("Cancel tx .* was confirmed at height '108'")
         wait_for(
             lambda: w.rpc.listvaults([], [deposit])["vaults"][0]["status"] == "canceled"
         )
 
+
 @pytest.mark.skipif(not POSTGRES_IS_SETUP, reason="Needs Postgres for servers db")
 def test_reorged_revault(revault_network, bitcoind):
     revault_network.deploy(4, 2, csv=12)
@@ -1533,20 +1547,18 @@ def test_reorged_revault(revault_network, bitcoind):
     fee = revault_network.compute_spendtx_fees(feerate, 1, 1)
     destinations = {addr: amount - fee}
     revault_network.unvault_vaults([vault], destinations, feerate)
-    unvault_tx = mans[0].rpc.listonchaintransactions([deposit])[
-        "onchain_transactions"
-    ][0]["unvault"]
+    unvault_tx = mans[0].rpc.listonchaintransactions([deposit])["onchain_transactions"][
+        0
+    ]["unvault"]
 
     # Now let's cancel the spending
     revault_network.cancel_vault(vault)
-
-    cancel_tx = mans[0].rpc.listonchaintransactions([deposit])[
-        "onchain_transactions"
-    ][0]["cancel"]
+    cancel_tx = mans[0].rpc.listonchaintransactions([deposit])["onchain_transactions"][
+        0
+    ]["cancel"]
 
     # Reorging, but not unconfirming the cancel
     bitcoind.simple_reorg(cancel_tx["blockheight"])
-
     for w in stks + mans:
         w.wait_for_logs(
             [
@@ -1556,10 +1568,8 @@ def test_reorged_revault(revault_network, bitcoind):
             ]
         )
 
-
     # Let's unconfirm the cancel and check that the vault is now in 'canceling' state
     bitcoind.simple_reorg(cancel_tx["blockheight"], shift=-1)
-
     for w in stks + mans:
         w.wait_for_logs(
             [
@@ -1568,15 +1578,14 @@ def test_reorged_revault(revault_network, bitcoind):
                 "Rescan of all vaults in db done.",
             ]
         )
-
     for w in stks + mans:
         wait_for(
-            lambda: w.rpc.listvaults([], [deposit])["vaults"][0]["status"] == "canceling"
+            lambda: w.rpc.listvaults([], [deposit])["vaults"][0]["status"]
+            == "canceling"
         )
 
     # Confirming the cancel again
-    bitcoind.generate_block(6)
-
+    bitcoind.generate_block(1, wait_for_mempool=1)
     for w in stks + mans:
         w.wait_for_log("Cancel tx .* was confirmed at height .*")
         wait_for(
@@ -1589,15 +1598,10 @@ def test_reorged_revault(revault_network, bitcoind):
         w.wait_for_log(f"Vault {deposit}'s Unvault transaction .* got unconfirmed")
 
     # Here we go canceling everything again
-    bitcoind.generate_block(6)
+    bitcoind.generate_block(1, wait_for_mempool=2)
     for w in stks + mans:
+        w.wait_for_log("Unvault transaction .* is now confirmed")
         w.wait_for_log("Unvault transaction at .* is now being canceled")
-        wait_for(
-            lambda: w.rpc.listvaults([], [deposit])["vaults"][0]["status"] == "canceling"
-        )
-
-    bitcoind.generate_block(6)
-    for w in stks + mans:
         w.wait_for_log("Cancel tx .* was confirmed")
         wait_for(
             lambda: w.rpc.listvaults([], [deposit])["vaults"][0]["status"] == "canceled"
```

Reply:
Pushed it myself